hudi-common/src/main/java/org/apache/hudi/common/table/log/BaseHoodieLogRecordReader.java

```diff
@@ -161,16 +152,14 @@ protected BaseHoodieLogRecordReader(HoodieReaderContext readerContext,
         .setBasePath(readerContext.getTablePath()).build();
     // load class from the payload fully qualified class name
     HoodieTableConfig tableConfig = this.hoodieTableMetaClient.getTableConfig();
-    this.payloadClassFQN = tableConfig.getPayloadClass();
+    this.payloadClassFQN = tableConfig.getAvroPayloadClass();
```

Comment:
It would be good to pass in the `RecordMergeConfig` instance for all places using the payload class, and get the payload class name from `RecordMergeConfig` instance directly now.  In the future, it's easy to switch from using payload to merger API or deprecate the code path.